src/builder.rs

```diff
@@ -949,54 +1204,63 @@ impl InProgressSignatures for PartiallyAuthorized {
 
 /// A heisen[`Signature`] for a particular [`Action`].
 ///
-/// [`Signature`]: redpallas::Signature
-#[derive(Debug)]
+/// [`Signature`]: VerSpendAuthSig
+#[derive(Debug, Clone)]
 pub enum MaybeSigned {
     /// The information needed to sign this [`Action`].
     SigningMetadata(SigningParts),
-    /// The signature for this [`Action`].
-    Signature(redpallas::Signature<SpendAuth>),
+    /// The versioned signature for this [`Action`].
+    Signature(VerSpendAuthSig),
 }
 
 impl MaybeSigned {
-    fn finalize(self) -> Result<redpallas::Signature<SpendAuth>, BuildError> {
+    fn finalize(self) -> Result<VerSpendAuthSig, BuildError> {
         match self {
             Self::Signature(sig) => Ok(sig),
             _ => Err(BuildError::MissingSignatures),
         }
     }
 }
 
-impl<P: fmt::Debug, V> Bundle<InProgress<P, Unauthorized>, V> {
+impl<Proof: fmt::Debug, V, P: OrchardPrimitives> Bundle<InProgress<Proof, Unauthorized>, V, P> {
```

Comment:
_From str4d ([r2602570477](https://github.com/zcash/orchard/pull/471#discussion_r2602570477))_

I would prefer that you do not rename existing generic parameter types. In particular, `Proof` is the name of the concrete proof type of an authorized bundle, so it was intentional that we did not use it here for a type that may or may not be a proof yet, to avoid confusion.
```suggestion
impl<P: fmt::Debug, V, Pr: OrchardPrimitives> Bundle<InProgress<P, Unauthorized>, V, Pr> {
```
(and similarly below)

Reply:
FIXED but only in these two spots:

* `src/builder.rs: impl<P: fmt::Debug, V, Pr: OrchardPrimitives> Bundle<InProgress<P, Unauthorized>, V, Pr> { ... }`
* `src/builder.rs: impl<P: fmt::Debug, V, Pr: OrchardPrimitives> Bundle<InProgress<P, PartiallyAuthorized>, V, Pr> { ... }`

That is, only where the generic name `P` conflicts with the existing `P` used for the proof type. We may also want to rename `P` to `Pr` in other impls that use `P: OrchardPrimitives`, to keep naming consistent.